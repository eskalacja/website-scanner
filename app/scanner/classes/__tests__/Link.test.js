const { Link } = require('../Link');

describe('classes/Link', () => {
  it('should create a root link', () => {
    const link = Link.fromString('https://example.com');
    expect(link).toMatchSnapshot();
  });

  it('should create a link with a parent', () => {
    const parent = Link.fromString('https://example.com');
    const link = Link.fromString('https://example.com/en', parent);

    expect(link).toMatchSnapshot();
  });
});
