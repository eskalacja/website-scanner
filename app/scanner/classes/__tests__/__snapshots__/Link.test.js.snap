// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classes/Link should create a link with a parent 1`] = `
Link {
  "error": undefined,
  "hash": "",
  "href": "https://example.com/en",
  "isChecked": false,
  "normalizedHref": "https://example.com/en",
  "origin": "https://example.com",
  "parents": Map {
    "https://example.com/" => Link {
      "error": undefined,
      "hash": "",
      "href": "https://example.com/",
      "isChecked": false,
      "normalizedHref": "https://example.com/",
      "origin": "https://example.com",
      "parents": Map {},
      "pathname": "/",
      "protocol": "https:",
      "search": "",
      "type": 1,
    },
  },
  "pathname": "/en",
  "protocol": "https:",
  "search": "",
  "type": 1,
}
`;

exports[`classes/Link should create a root link 1`] = `
Link {
  "error": undefined,
  "hash": "",
  "href": "https://example.com/",
  "isChecked": false,
  "normalizedHref": "https://example.com/",
  "origin": "https://example.com",
  "parents": Map {},
  "pathname": "/",
  "protocol": "https:",
  "search": "",
  "type": 1,
}
`;
